sw/device/silicon_creator/lib/ownership/isfb.c

```diff
@@ -0,0 +1,122 @@
+// Copyright lowRISC contributors (OpenTitan project).
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/silicon_creator/lib/drivers/flash_ctrl.h"
+#include "sw/device/silicon_creator/lib/error.h"
+#include "sw/device/silicon_creator/lib/manifest.h"
+#include "sw/device/silicon_creator/lib/ownership/owner_block.h"
+
+enum {
+  kFlashWordSize = 64,
+
+  // The maximum number of product expressions is 256, which maps to 512 flash
+  // words or 1024 bytes.
+  kProductExprMaxCount = 256,
+
+  // The maximum size of the product expressions in bytes.
+  KProductExprMaxCountBytes = kProductExprMaxCount * sizeof(uint32_t),
+
+  // The number of bits in the strike mask is fixed to 256. Each bit in the
+  // strike mask corresponds to a `uint32_t` word.
+  kExpectedStrikeBitCount = 256,
+
+  // The expected size of the strike region in bytes and words.
+  kExpectedStrikeRegionBytesCount = kExpectedStrikeBitCount * sizeof(uint32_t),
+  kExpectedStrikeRegionWordsCount =
+      kExpectedStrikeRegionBytesCount / sizeof(uint32_t),
+};
+
+rom_error_t isfb_boot_request_process(const manifest_ext_isfb_t *ext,
+                                      const owner_config_t *owner_config,
+                                      uint32_t *checks_performed_count) {
+  *checks_performed_count = UINT32_MAX;
+  if (ext->header.name != kManifestExtIdIsfb ||
+      (hardened_bool_t)owner_config->isfb == kHardenedBoolFalse) {
+    return kErrorOwnershipISFBNotPresent;
+  }
+  const owner_isfb_config_t *isfb = owner_config->isfb;
+  if (ext->product_expr_count * 2 > isfb->product_words ||
+      ext->product_expr_count > kProductExprMaxCount) {
+    return kErrorOwnershipISFBProductExpCnt;
+  }
+
+  // The ISFB info page configuration is handled in the owner's
+  // `FlashInfoConfig` settings.
+  flash_ctrl_info_page_t isfb_info_page;
+  HARDENED_RETURN_IF_ERROR(flash_ctrl_info_type0_params_build(
+      isfb->bank, isfb->page, &isfb_info_page));
```

Comment:
We should probably have a check somewhere to make sure that the owner FlashInfoConfig exists and has a configuration for this bank and page.

Reply:
Let me add this check in a follow up PR. I need to think about how to implement checks that rely on other `TLV` entries.

Reply:
Ack.